internal/aggregator/utils_test.go

```diff
@@ -74,3 +82,199 @@ func TestGetExpectedTPUNodePoolSize(t *testing.T) {
+												k8sutils.NodeLabelGKESpot:           "true",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: records.Attrs{
+				JobSetName:      "test-jobset",
+				JobSetNamespace: "test-ns",
+				JobSetUID:       "12345",
+				TPUAccelerator:  "tpu-v5p",
+				TPUTopology:     "2x2x1",
+				Spot:            true,
+				TPUChipCount:    4,
+			},
+		},
+		"multiple replicated jobs": {
+			js: &jobset.JobSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-jobset",
+					Namespace: "test-ns",
+					UID:       "12345",
+				},
+				Spec: jobset.JobSetSpec{
+					ReplicatedJobs: []jobset.ReplicatedJob{
+						{
+							Replicas: 1,
+							Template: batchv1.JobTemplateSpec{
+								Spec: batchv1.JobSpec{
+									Template: corev1.PodTemplateSpec{
+										Spec: corev1.PodSpec{
+											NodeSelector: map[string]string{
+												k8sutils.NodeLabelGKETPUAccelerator: "tpu-v5p",
+												k8sutils.NodeLabelGKETPUTopology:    "2x2x1",
+												k8sutils.NodeLabelGKESpot:           "false",
+											},
+										},
+									},
+								},
+							},
+						},
+						{
+							Replicas: 2,
+							Template: batchv1.JobTemplateSpec{
+								Spec: batchv1.JobSpec{
+									Template: corev1.PodTemplateSpec{
+										Spec: corev1.PodSpec{
+											NodeSelector: map[string]string{
+												k8sutils.NodeLabelGKETPUAccelerator: "tpu-v5p",
+												k8sutils.NodeLabelGKETPUTopology:    "2x4x1",
+												k8sutils.NodeLabelGKESpot:           "false",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: records.Attrs{
+				JobSetName:      "test-jobset",
+				JobSetNamespace: "test-ns",
+				JobSetUID:       "12345",
+				TPUAccelerator:  "tpu-v5p",
+				TPUTopology:     "2x4x1",
+				Spot:            false,
+				TPUChipCount:    20,
+			},
+		},
+		"missing topology": {
+			js: &jobset.JobSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-jobset",
+					Namespace: "test-ns",
+					UID:       "12345",
+				},
+				Spec: jobset.JobSetSpec{
+					ReplicatedJobs: []jobset.ReplicatedJob{
+						{
+							Replicas: 1,
+							Template: batchv1.JobTemplateSpec{
+								Spec: batchv1.JobSpec{
+									Template: corev1.PodTemplateSpec{
+										Spec: corev1.PodSpec{
+											NodeSelector: map[string]string{
+												k8sutils.NodeLabelGKETPUAccelerator: "tpu-v5p",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: records.Attrs{
+				JobSetName:      "test-jobset",
+				JobSetNamespace: "test-ns",
+				JobSetUID:       "12345",
+				TPUAccelerator:  "tpu-v5p",
+				Spot:            false,
+				TPUChipCount:    0,
+			},
+		},
+	}
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			got := extractJobSetAttrs(c.js)
+			if diff := cmp.Diff(c.want, got); diff != "" {
```

Comment:
Elsewhere we use the `github.com/stretchr/testify` library for doing this, any reason to use `cmp` here instead?

Reply:
habit :)

updated to use testify in 1f80882